pact/policy-manager/policy-manager.pact

```diff
@@ -361,45 +370,65 @@
              (sale-type:string (at 'sale-type quote-spec))
              (sale-price:decimal (at 'sale-price quote-spec))
            )
-      (if (!= sale-type "")
-        (if (exists-msg-decimal UPDATED-PRICE-KEY)
-          (let ((updated-price:decimal (read-msg UPDATED-PRICE-KEY)))
-            (fungible::enforce-unit updated-price)
-            (with-capability (UPDATE-QUOTE-PRICE (at 'id token) sale-id sale-type updated-price)
-              (update quotes sale-id { "sale-price": updated-price })
+
+        ; If a sale contract is being used, update the price accordingly
+        (if (!= sale-type "")
+          (if (exists-msg-decimal UPDATED-PRICE-KEY)
+            (let ((updated-price:decimal (read-msg UPDATED-PRICE-KEY)))
+              (fungible::enforce-unit updated-price)
+              (with-capability (UPDATE-QUOTE-PRICE (at 'id token) sale-id sale-type updated-price)
+                (update quotes sale-id { "sale-price": updated-price })
+              )
+              true
             )
-            true
+          true
           )
-         true
+          true
         )
-        true
-      )
-      (let ((final-sale-price:decimal  (at 'sale-price (get-quote-info sale-id))))
+
+        (let* (
+            (final-sale-price:decimal  (at 'sale-price (get-quote-info sale-id)))
+            (mk-fee-spec:object{marketplace-fee-spec} (try { "mk-account": "", "mk-fee-percentage": 0.0 } (read-msg MARKETPLACE-FEE-KEY)))
+            (mk-fee:decimal (floor (* (at 'mk-fee-percentage mk-fee-spec) final-sale-price) (fungible::precision)))
```

Comment:
We should enforce that the percentage is >= 0.0 (definite), and we could also enforce it's below 1.0 (open question, should we leave it to users? )

Reply:
It adheres to the same structure as the royalty policy, and I see that we're doing the validation there so let's do it here as well. Will add it.